docs/DeveloperGuide.md

```diff
@@ -1,37 +1,1040 @@
+`DayList#isPlanValid` methods respectively.
+
+(Steps 7.9 and 7.11) If the day number or plan number is not valid, an `InvalidScheduleException` would be thrown to the user,
+and the entire updating of a plan for a particular day in the schedule process is aborted.
+
+(Steps 7.13 and 7.14) `PlanList#getPlanDisplayList` method is called to find and return the hash value of the plan number, 
+`planToAddKey:String` to be scheduled for a particular day. The `planToAddKey` is used to get the `plan` object in the 
+planList by calling the `PlanList#getPlanFromKey` (steps 7.15 to 7.16).
+
+(Steps 7.17 to 7.18) Once the `Plan` object is retrieved, if there are no plan being scheduled for the day, 
+a new Day object is created and stored in the application.
+
+(Steps 7.19 to 7.20) However, if there is an existing plan scheduled for that particular day, the `Day` object that 
+had already been created, will then be updated to store the latest plan scheduled for the day.
+
+(Step 9 and 10) After successfully created/updated the Day object, the `UI#printNewSchedule(newDay)` method
+will be called to display the day and the corresponding plan scheduled for it via the terminal. The following is an
+example of the message after the user had successfully scheduled a plan for the day (e.g. `schedule /update 1 1`):
+```
+----------------------------------------------------------------------
+Alright, the following plan schedule has been created:
+
+Monday -- arms
+
+----------------------------------------------------------------------
+```
+(Step 11) Lastly, before the `ScheduleCommand` object is discarded, the `FileManager#rewriteAllDaysScheduleToFile(dayList)`
+is called to rewrite the `schedule.txt` file according to the newly modified application's day list.
+
+This completes the process of scheduling a plan for a particular day in WerkIt!
+
+##### Design considerations for update schedule
+###### Day Object
+For the application, schedule is defined to be a 7-days workout plan. The days that do not have any plan scheduled
+would be considered a rest day for the user. Therefore, when implementing the creation of `Day` object, a total of 7
+`Day` objects at most would be created and be stored in the dayList with size 7.
+
+Initially, if no plan has been scheduled for a particular day, the corresponding Day object would not be created.
+For example, if no plan is being scheduled for Monday, there will be no Day object created for Monday and the dayList
+with index 0 will not have any `Day` object being stored.
+
+If `dayList[0]` contains a `Day` object, it would mean that the user scheduled a plan on Monday. If the
+user were to execute the `schedule /update` command again to update the plan to be scheduled for Monday, the application
+will update the content in the Day object stored in `dayList[0]`. It will not recreate a Day object for Monday
+to store the new plan.
```

Comment:
Somehow, this part of the DG is completely bolded and enlarged. It could be that this is to highlight a certain concept to be cautious of, but it seems to stand out more than the headers and other parts of the DG that might take precedence in terms of the structure of the DG. perhaps this could be written in Italics, and keywords be bolded.

![image](https://user-images.githubusercontent.com/26668506/161182763-cf2ed39e-3814-48e2-a3e6-5192b2459d6f.png)